rmw_dds_common/src/qos.cpp

```diff
@@ -374,4 +374,62 @@ qos_profile_check_compatible(
   return RMW_RET_OK;
 }
 
+rmw_ret_t
+qos_profile_get_most_compatible_for_subscription(
```

Comment:
should this function also take in accound `deadline`?

You need `deadline_publisher < deadline_subscription` for them to match